crates/iota-indexer/src/indexer.rs

```diff
@@ -98,6 +98,19 @@ impl Indexer {
             spawn_monitored_task!(pruner.start(CancellationToken::new()));
         }
 
+        if let Some(optimistic_pruner_batch_size) = pruning_options.optimistic_pruner_batch_size {
+            info!("Starting indexer optimistic tables pruner");
+            let optimistic_pruner: OptimisticPruner = OptimisticPruner::new(
```

Comment:
I suppose we can remove the type in the variable declaration since the `OptimisticPruner::new` does not return a generic type but `Self`.

```suggestion
            let optimistic_pruner = OptimisticPruner::new(
```